doc/performance-investigations/0009-eq-session-datastore-over-redis/summary.md

```diff
@@ -0,0 +1,58 @@
+# eq-session in Datastore instead of Redis
+
+Recent scale and stress testing has shown that using Memorystore Redis as the backend for eq-session with the current Runner and Redis architecture does not support the volume of requests required for upcoming Runner use cases.
+
+This test is to investigate the impact of moving eq-session from Redis to Datastore under the current Runner v3.52.0 implementation and web server architecture (gunicorn-async with 7 workers)
+
+https://github.com/ONSdigital/eq-questionnaire-runner/tree/performance-investigation-eq-session-datastore
+
+## Benchmark profile
+
+| Option                 | Value                        |
+| ---------------------- | ---------------------------- |
+| Requests file          | census_household_gb_eng.json |
+| Run time               | 20m                          |
+| User wait time minimum | 1                            |
+| User wait time maximum | 2                            |
+| Clients                | 200                          |
+| Hatch rate             | 200                          |
+| Locust Instances       | 2                            |
+
+## Results
+
+Locust results based on 99th percentile timings.
+
+| Metric | Baseline | Investigation |
+|--------|----------|--------------|
+| Questionnaire GETs | 333ms | 340ms|
+| Questionnaire POSTs | 374ms| 379ms|
+| All requests | 353ms | 359ms |
+| Total requests (over 20 mins) | 750,432 | 746,720 |
+
+Environment monitoring observations
+
+- 10 Runner instances (30 cores)
```

Comment:
Is there a specific reason we chose to do 10 instances instead of mimicking the daily test set up?. As I thought we planned to do all these with similar structure for a better comparison.

Reply:
The comparison is against a benchmark for any individual investigation. We shouldn't infer anything for trying to compare against previous investigations (or the daily test). In this particular case we discussed and agreed to use 10 instances as it would also give an indication of the impact at slightly higher loads. The load injectors were increased in order to maintain a similar (albeit slightly higher) volume of requests over this number of instances.

The difference in latencies is only part of the picture though. The amount of CPU to serve those requests (even at higher latencies) also increases and 10 instances was intended to give a better indication of what that impact was.

If we feel we should always run these in exactly the same config as the daily tests to be able to compare not only to the benchmark but also across investigations we can discuss that and capture it as a rule for future tests.

Reply:
It wasn't to compare to existing benchmark but it was to make things more controlled between different tests, hence why we said we would use daily test set up for https://trello.com/c/XqqqZvuh/4217-gunicorn-performance-benchmark-m

As far as the investigation is concerned I agree the comparison is against the baseline and it is enough to highlight what to do next.